core/app/models/spree/order.rb

```diff
@@ -516,6 +516,8 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
+      return self.shipments if unreturned_exchange? && shipments.all?(&:shipped?)
```

Comment:
The spec that covers why it's required is in `unreturned_item_charger_spec.rb`. It's a though one to bring over to `order_spec.rb` since it's directly related to this method being called when attempting to charge for unreturned exchanges.

I think raising might be too severe since it's not an exceptional case, perhaps wrapping the contents of `create_proposed_shipments` within an `unless unreturned_exchange? && shipments.all?(&:shipped?)` would make it a little clearer?